include/multipass/virtual_machine.h

```diff
@@ -59,6 +59,7 @@ class VirtualMachine
     virtual std::string ipv4() = 0;
     virtual std::string ipv6() = 0;
     virtual void wait_until_ssh_up(std::chrono::milliseconds timeout) = 0;
+    virtual void ensure_vm_is_running(){};
```

Comment:
Hmm, so this is just a way to detect whether or not we are in a mock? Not sure that is advisable... And having the implementation here drops the pure interface.

I would prefer keeping this an interface and having a pure virtual `wait_for_cloud_init` here. Then an intermediate descendant (i.e. `VirtualMachineBase` or something) to implement it, and then real implementations inheriting from that one. The mock one could then throw or do whatever appropriate. If we later need to move to the alternative *composition-over-inheritance* approach that would be easy.

Reply:
No, this is not used a just a way to detect whether we are in a mock or not.  It's used to ensure the particular backend's instance is actually running.  I had to make it public so utils can call it.  But it's been private before.  If we want this to remain an interface, then I can make `ensure_vm_is_running` a pure virtual since it's needed by all the backends any ways.  Does that work for you?

I don't think it's necessary to have `wait_for_cloud_init` as a pure virtual since everything that is needed by that call can be handled via the utils call.

Reply:
Oh you're right! I looked only at the diff and thought `ensure_vm_is_running` had been reintroduced now, but then only saw an empty implementation in the base and a throwing one in the stub... fogettaboudit 

If you could just keep it pure virtual then, to make it more tidy...

Reply:
Ok, `ensure_vm_is_running` is now a pure virtual.